draft-ietf-lamps-dilithium-certificates.md

```diff
@@ -747,6 +708,110 @@ previous section.
+additional discussion of why HashML-DSA is disallowed in PKIX.
+
+All functions and notation used in {{fig-externalmu-ml-dsa-external}}
+and {{fig-externalmu-ml-dsa-internal}} are defined in {{FIPS204}}.
+
+External operations:
+
+~~~
+ExternalMu-ML-DSA.Prehash(pk, M, ctx):
+
+  if |ctx| > 255 then
+    return error  # return an error indication if the context string is
+                  # too long
+  end if
+
+  M' = BytesToBits(IntegerToBytes(0, 1) ∥ IntegerToBytes(|ctx|, 1)
+                                                        || ctx) || M
+  mu = H(BytesToBits(H(pk, 64)) || M', 64)
+  return mu
+~~~
+{: #fig-externalmu-ml-dsa-external title="External steps of ExternalMu-ML-DSA"}
+
+Internal operations:
+
+~~~
+ExternalMu-ML-DSA.Sign(sk, mu):
+
+  if |mu| != 512 then
+    return error  # return an error indication if the input mu is not
+                  # 64 bytes (512 bits).
+  end if
+
+  rnd = rand(32)  # for the optional deterministic variant,
+                  # set rnd to all zeroes
+  if rnd = NULL then
+    return error  # return an error indication if random bit
+                  # generation failed
+  end if
+
+  sigma = ExternalMu-ML-DSA.Sign_internal(sk, mu, rnd)
+  return sigma
+
+ExternalMu-ML-DSA.Sign_internal(sk, mu, rnd): # mu is passed as argument instead of M'
+   ... identical to FIPS 204 Algorithm 7, but with Line 6 removed.
+~~~
+{: #fig-externalmu-ml-dsa-internal title="Internal steps of ExternalMu-ML-DSA"}
+
+ExternalMu-ML-DSA requires the public key, or its prehash, as input to
+the pre-digesting function. This assumes the signer generating the
+pre-hash is in possession of the public key before signing and is
+different from conventional pre-hashing which only requires the
+message and the hash function as input.
+
+Security-wise, during the signing operation of pure (or "one-step")
+ML-DSA, the cryptographic module extracts the public key hash `tr` from
+the secret key object, and thus there is no possibility of mismatch
+between `tr` and `sk`. In ExternalMu-ML-DSA, the public key or its hash
+needs to be provided to the `Prehash()` routine indpedendly of the secret
+key, and while the exact mechanism by which it is delivered will be
+implementation-specific, it does open a windown for mismatches between
+`tr` and `sk`. First, this will produce a signature which will fail to
+verify under the intended public key since a compliant `Verify()` routine
+will independently compute `tr` from the public key. Second, a malicious
+or tricked signer could potentially produce a signature which validates
+under a different public key, which weakens the implicit security
+assumptions of the ML-DSA algorithm. Implementors should pay careful
+attention to how the public key or its hash is delivered to the
+`ExternalMu-ML-DSA.Prehash()` routine, and from where they are sourcing
+this data. Note that HashML-DSA also weakens this security assumption even
```

Comment:
Remove the mention to HashML-DSA, it is covered already in the Security Considerations.